bundles/mapping/layerswipe/instance.js

```diff
@@ -71,14 +72,21 @@ Oskari.clazz.define(
                 this.unregisterEventListeners();
                 this.hideSplitter();
                 Oskari.getSandbox().getService('Oskari.mapframework.service.VectorFeatureService').setHoverEnabled(true);
+                this.mapModule.setSwipeStatus(false, null, null);
             }
             this.active = active;
             this.mapModule.getMap().render();
         },
 
         updateSwipeLayer: function () {
             this.unregisterEventListeners();
-            this.layer = this.getTopmostLayer();
+            const topLayer = this.getTopmostLayer();
+            this.layer = topLayer.ol;
+            this.oskariLayer = topLayer.layerId;
+
+            if (this.oskariLayer !== null) {
+                this.mapModule.setSwipeStatus(true, this.cropSize, this.oskariLayer);
```

Comment:
Since the map module doesn't need this status for itself and it's already quite big and complex, I would rather store this on the layer itself (`oskariLayer`). Then we only need to store the `cropSize`. It's currently named as `x` on the status, but I'm thinking it might be more clear if we call it something like something like partiallyHidden or "partial visibility" with an object with keys `right` or `left` or something instead of "swipe". 

Also I would rather have the value in one place instead of having both "cropSize" on the layerswipe AND whatever it ends up on the layer/mapmodule. With this the same value is stored in two places which is error prone. We could remove cropSize from the swipe tool and store it on the layer itself as well.

Any thoughts on this?

Reply:
Another thought is that we could save the "swipe status" on the gfi-plugin and skip setting it on the map module. Then it would be almost identical but it would make more sense to store it at the gfi-plugin. Storing the status on the layers would make it harder to tear down the status I suppose since when layers are switched you should remove the partial visibility from one and add it to another.

Reply:
Swipe status moved to the plugin now.